sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/functions/FunctionsTest.java

```diff
@@ -349,6 +386,19 @@ public void testDecimalErrors() {
                 // The compiler rounds `1234.1236` to `1234.124` before calling the rust code
                 "cannot represent 1234.124 as DECIMAL(6, 3)"
         );
+
+        this.shouldFail("select cast(1234.1234 AS DECIMAL(6, 3))",
+                "cannot represent 1234.123 as DECIMAL(6, 3)"
+        );
+
+        this.shouldFail("select cast(1234.1236 AS DECIMAL(6, 3))",
+                "cannot represent 1234.123 as DECIMAL(6, 3)"
+        );
+
+        // this only fails in runtime
+        this.shouldFail("select cast(143.481 as decimal(2, 1))", "cannot represent 143.4 as DECIMAL(2, 1)");
```

Comment:
Is this message correct? The literal seems to have the wrong value. It's already missing two digits.
Maybe it has been truncated already.
If we could fix that it would be good, this can be very confusing for users.

Reply:
For compile time errors, the ones we give during simplification, we can use the literal, to print the actual value in the error.